src/main/java/com/PBA/budgetservice/controller/BudgetController.java

```diff
@@ -0,0 +1,47 @@
+package com.PBA.budgetservice.controller;
+
+import com.PBA.budgetservice.persistance.model.dtos.IncomeRequest;
+import com.PBA.budgetservice.persistance.model.dtos.IncomeResponse;
+import com.PBA.budgetservice.persistance.model.dtos.IncomeUpdateRequest;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.responses.ApiResponses;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/api/budget")
+public interface BudgetController {
+    @Operation(summary = """
+            Creates an income corresponding to the user uid specified in an IncomeRequest request body.
+            """)
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "201", description = "Created")
+    })
+    @PostMapping("/income")
+    public ResponseEntity<Void> createIncome(
+            @io.swagger.v3.oas.annotations.parameters.RequestBody(description = "Income to create")
+            @RequestBody IncomeRequest incomeRequest);
+
+    @Operation(summary = """
+            Provides a list of IncomeResponse entities that contain all the data of all the incomes currently stored in the application.
```

Comment:
Build the summary to be understood by everyone. What is IncomeResponse entities? Have a natural language, like "Get all incomes from the system."
The summary should be very short and concise, but the description can be bigger.